src/request/creditcard.ts

```diff
@@ -0,0 +1,84 @@
+import { ArgumentError } from '../errors';
+
+interface CreditCardProps {
+  issuerIdNumber?: string;
+  last4digits?: string;
+  token?: string;
+  bankName?: string;
+  bankPhoneCountryCode?: string;
+  bankPhoneNumber?: string;
+  avsResult?: string;
+  cvvResult?: string;
+}
+
+const singleChar = /^[A-Za-z0-9]$/;
+const issuerIdNumberRegex = /^[0-9]{6}$/;
+const last4Regex = /^[0-9]{4}$/;
+const tokenRegex = /^(?![0-9]{1,19}$)[\u0021-\u007E]{1,255}$/;
+
```

Comment:
A general thought about testing these regexes. Some of these regexes are complex and have specific rules, which make me think that we should tests them more. Perhaps we can use `jest.each` to iterate through a list of values supplied to the existing tests you wrote. For each value, we can specify if we expect the value have an error thrown or not.